src/WebJobs.Script.WebHost/WebJobsApplicationBuilderExtension.cs

```diff
@@ -71,10 +70,7 @@ public static IApplicationBuilder UseWebJobsScriptHost(this IApplicationBuilder
             {
                 config.UseMiddleware<HomepageMiddleware>();
             });
-            builder.UseWhen(context => HttpThrottleMiddleware.ShouldEnable(httpOptions.CurrentValue) && !context.Request.IsAdminRequest(), config =>
```

Comment:
Unlike some of the other options types being used to conditionally register other middleware in this method, HttpOptions is ScriptHost scoped, so it can't be used here. I.e. on startup when this code runs, it has the default value, and will never receive the value read from host.json.